server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -291,6 +294,7 @@ private Sink getSink(long timestamp)
 
                         // The realtime plumber always uses SingleElementPartitionChunk
                         final Sink theSink = holder.getObject().getChunk(0).getObject();
+                        final boolean skipIncrementalSegment = query.getContextValue(SKIP_INCREMENTAL_SEGMENT, false);
```

Comment:
I think we only support Strings for backwards compatibility, do we have to support strings for new properties ?